### PR TITLE
Use type annotations to constrain integer solutions.

### PR DESCRIPTION
**Pull request by joelburget**
In the case a dataclass member is annotated as an int, we can tell sympy
to generate only integer solutions for it. For now, we assume anything
that's not an int is a float.

### CODE REVIEW

**Comment by joelburget** [comment]
Hi Daniel, a couple of notes about this change:

* You'll notice I removed `solve`, since it's not used anymore.
* Actually, `System` is also no longer used by the wrapper. It wasn't clear to me if the `System` class has some capability that my use of sympy doesn't. If so we should add a test for that functionality.

Please let me know if I got either of these wrong.

File: constraintula/core.py [hunk 1]
```diff
@@ -210,12 +212,22 @@ def get_functions(self) -> Mapping[Symbol, Callable]:
         return functions
 
 
-def solve(
-    independents: Mapping[str, Any], variables: Sequence[str], constraints: Iterable[Expr]
-) -> Mapping[str, Any]:
-    assignments = {Symbol(k): v for k, v in independents.items()}
-    values = System(constraints).with_independents(assignments).evaluate(assignments)
-    return {k: values[Symbol(k)] for k in variables}
+def collect_symbols(expr):
```
Comment (DanielSank) [inline constraintula/core.py]: Type annotations?

**Review by DanielSank — COMMENTED** [review]
This looks really nice. Thanks! I have only one small nitpick.

File: constraintula/core.py (hunk 1 above)
Comment (joelburget) [reply constraintula/core.py]: I added annotations but I wasn't sure how to check them. What command do you use?

Comment (DanielSank) [reply constraintula/core.py]: Well, the simplest thing is to pip install mypy and just run it against the target .py file. We could set up a script to run against all source files.

**Review by DanielSank — APPROVED** [review]

**Pull request merged**